Public/Src/Cache/ContentStore/Vsts/DedupReadOnlyContentSession.cs

```diff
@@ -178,6 +178,15 @@ private enum Counters
                 return pinResult;
             }
 
+            VstsBlobIdentifier blobId = ToVstsBlobIdentifier(contentHash.ToBlobIdentifier());

```

Comment:
Would it make sense to make the optimization more effecient by checking `Contenthash.HashType`?

i.e. 
```
if (contentHash.HashType == HashType.DedupChunk)
```

now we are converting and deserializing just to check and not using blobId or dedupId in the fast path...

Reply:
Not sure what we would be saving, since what you're suggesting still requires us to do the conversion every single time.